alerts/cluster-monitoring-operator/ThanosRuleEvaluationLatencyHigh.md

```diff
@@ -0,0 +1,46 @@
+# ThanosRuleEvaluationLatencyHigh
+
+## Meaning
+
+The [Thanos Ruler][1] is deployed when user workload monitoring is enabled. It
+allows alerting rules deployed as part of user workload monitoring to query
+both the Prometheus instance responsible for cluster components as well the
+user-workload Prometheus instance. This alert is triggered when the
+Thanos Ruler takes longer time to evaluate the alert.
+
+## Impact
+
+Alerts will take longer time to get delivered.
+
+## Diagnosis
+
+Check the logs for the Thanos Ruler pods:
+
+```console
+oc -n openshift-user-workload-monitoring logs -l 'thanos-ruler=user-workload'
+-c thanos-ruler
+```
+
+Check the logs for the Thanos Querier pods:
+
+```console
+oc -n openshift-monitoring logs -l 'app.kubernetes.io/name=thanos-query'
+-c thanos-query
+```
+
+This alert will trigger when the rule evaluation take more time than
+the scheduled interval. It can indicate that your query backend
+(i.e Thanos Querier) takes too much time to evaluate the query. If there is
+even a single rule that is taking too long (more than the interval for that
+group) to evaluate, this alert will fire. This might indicate other problems
+like slow StoreAPIs or too complex query expression in rule.
+
+## Mitigation
```

Comment:
I would mention a few more things:
* verify if resource limits are set on the different monitoring components and whether they are throttled.
* Audit the rule groups that fire the alert to identify expensive queries
* Consider splitting the rule groups into smaller groups if possible.